### DISCUSSION TITLE
Clarify behavior of non stale, cached data on app reopen, after error in queryFn

### DISCUSSION THREAD

**Discussion by sak3th** [TanStack/query#4172] [Q&A]
_Version: 4.2.3_

**Scenario**: Use `useQuery` to fetch some data with  infinite stale and cache time. 

- Step 1 -  Kill the app and make sure that the `queryFn` from this moment onwards always errors out.
- Step 2 - Start the app. (`queryFn ` is triggered with `useQuery` on mount). Data is served from cache as there is an error in the `queryFn`. Triggering `refetch` also serves cached data.
- Step 3 -  Kill the app and start it again.
- Step 4 - This time cached data is not getting served and data comes as `undefined`

**Question** - **After Step 3, somehow the cached data was invalidated even though the stale time was set to `Infinity`. Why is the cache data not being served from 2nd app restart and when is the cache data getting invalidated?**

(I tried to get around the problem/behavior by using `initialData` and `initialDataUpdatedAt` but was not able to make it work.)

**Comment by TkDodo** [comment]
You have to be more precise please. Talking about "app" - is this react native? How is data still there after you "killed the app"? So you use persisters?

**Reply by sak3th** [reply]
Yes. This is react native and we are using persisters.

Let me know if there is any other info required.

**Reply by TkDodo** [reply]
Yes. A reproduction.
